### ISSUE TITLE
[Bug] Pandoc process slowly consumes all RAM and never completes with PDF (Latex) or HTML project export

### ISSUE THREAD

**Issue by cbop-dev** [Zettlr/Zettlr#3217]
### Description

With version 2.2.1 for Linux (on Mint Cinnamon 20.3, using Zettlr amd64.deb package), I cannot complete an export of any project into PDF (Latex) or HTML. The pandoc process runs at 100% CPU and the RAM grows indefinitely -- even up to 16GB! -- and must be force-quit.

This happens even for the simplest of projects: one file, with simple header and one sentence of text. 

This also happens with and without a custom template, and with or without the pre-packaged `pandoc` binary. In the meantime, I must revert to version 2.1.3. When I revert to 2.1.3, with the exact same settings, the problem goes away and I can export to PDF (latex) immediately.

_Nota bene_: this does not happen when exporting a single file using the export button. When I export a project .md file to PDF (Latex) this way, I do not see a `pandoc` process appear, but only a `xelatex` process.

### Reproducing

1. Create a new workspace and a new project directory in the workspace. 
2. Set project settings to compile to PDF (Latex) or HTML.
2. Create one simple markdown file in the project directory. 
3. Compile the project.
4. Watch the pandoc process slowly consume all system memory (e.g. with `top -bc | grep pandoc`). Force quit to stop (`killall pandoc`).
4. Try again with (or without) custom template. Same results.
5. Try again with and without packaged pandoc version.  Same results.
6. Uninstall and revert to Zettlr 2.1.3. 
7. Compile the same project without any problems. 
8. Wait patiently for bug fix while you use an out-of-date version of Zettlr. :-(

### Zettlr Version

Stable (most recent version)

### Specify version

2.2.1

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Linux Mint 20.3 Cinnamon

**Comment by nathanlesage** [comment]
> When I revert to 2.1.3, with the exact same settings, the problem goes away and I can export to PDF (latex) immediately.

This indicates a potential problem with the Pandoc version that shipped with 2.2.0. If I remember correctly, there have been a few updates in the meantime, so potentially this problem will go away with the next update to Zettlr.

**Comment by cbop-dev** [comment]
Except that the above problem still exists even when I disable "use internal version of Pandoc" and used, instead, my own locally installed version--and I've tried it with both 2.17.0.1 and 2.17.1.1. In both cases, using the local version of pandoc works fine with Zettlr 2.1.3, but not with v. 2.2.1. So the problem doesn't seem to be the packaged ("internal") version of pandoc, but the way the app is using/invoking pandoc, of whatever version. Unless, of course, the "use internal version of Pandoc" setting itself isn't working right.

**Comment by nathanlesage** [comment]
> So the problem doesn't seem to be the packaged ("internal") version of pandoc

That is right. So probably I made an error *again* in the project settings …

could you share a minimal reproducible template? Doesn't need to be some zip file, if one Markdown file with certain settings suffices!

**Comment by cbop-dev** [comment]
Sure, but it's literally just the simplest markdown file: one-word header, one paragraph with two words:

[20220215133747.md](https://github.com/Zettlr/Zettlr/files/8075662/20220215133747.md)

**Comment by nathanlesage** [comment]
mhh, I do see strange behavior; Zettlr does export, but apparently empty files o.O

**Comment by nathanlesage** [comment]
*Update, just had a look into top, and indeed, Pandoc is going crazy. Possibly a misconfiguration :S 

I'll have to look into it, but this week I'm booked with a course and a workshop, so not before the weekend :(

**Comment by cbop-dev** [comment]
I think I've discovered the situation in which this occurs, one source of the problem, and a simple workaround before there's a bug fix: 

**Problem:** By default, when exporting a project, the `defaults.yaml` file passed to `pandoc` lists _all files_ in the project folder as input files, _including PDFs and HTML files._ If the _output_ file (e.g., PDF or HTML) is also already in that folder (say, from a previous export), it is also included as _an input file_, _and as the output file._ This, I think, is creating a feedback loop of some sort, which can never finish, as it is treating the same file as both input and output.

**Workaround**: Apply a glob filter to the project folder (Project settings -> Files) to only include `*.md` files. When this is set, the problem disappears, because the output file is no longer also an input file.

I found this by looking at the generated `/tmp/defaults.yaml` that was being passed to the `pandoc` process. Running `pandoc --defaults /tmp/defaults.yaml` with this same file reproduces the problem when the output file is also in the input section. Removing that line from the yaml file removed the problem.

So, the fix may be to simply check and exclude the output file from the input section of the generated yaml file passed to `pandoc`.
